packages/opentelemetry-instrumentation-fetch/src/fetch.ts

```diff
@@ -355,6 +368,21 @@ export class FetchInstrumentation extends InstrumentationBase<
     };
   }
 
+  private _applyAttributesAfterFetch(
+    span: api.Span,
+    request: Request | RequestInit,
+    result: Response | FetchError
+  ) {
+    if (this._getConfig().applyCustomAttributesOnSpan) {
```

Comment:
```suggestion
    const applyCustomAttributesOnSpan = this._getConfig().applyCustomAttributesOnSpan;
    if (typeof applyCustomAttributesOnSpan === 'function') {
     safeExecuteInTheMiddle(
        () =>
          applyCustomAttributesOnSpan(span, request, result),
        () => {},
        true
      ); 
    }
```

You will have bullet proof validation and you can probably get rid of `!` from line 379

Reply:
nice